# HG changeset patch
# User Steve Peters <scpeters@osrfoundation.org>
# Date 1430782295 25200
# Branch issue_1074
# Node ID 05b7ee5db6f524e19b6cca54d66208cb48a0854c
# Parent  2cbd32c6246166f6f50af3503553180e7c1cd4a4
Fix build when compiled against bullet 2.83

diff --git a/gazebo/physics/bullet/BulletHinge2Joint.cc b/gazebo/physics/bullet/BulletHinge2Joint.cc
--- a/gazebo/physics/bullet/BulletHinge2Joint.cc
+++ b/gazebo/physics/bullet/BulletHinge2Joint.cc
@@ -211,8 +211,12 @@
     return math::Angle();
   }
 
-  btRotationalLimitMotor *motor =
-    this->bulletHinge2->getRotationalLimitMotor(_index);
+#ifndef LIBBULLET_VERSION_GT_282
+  btRotationalLimitMotor
+#else
+  btRotationalLimitMotor2
+#endif
+    *motor = this->bulletHinge2->getRotationalLimitMotor(_index);
   if (motor)
     return motor->m_hiLimit;
 
@@ -229,8 +233,12 @@
     return math::Angle(0.0);
   }
 
-  btRotationalLimitMotor *motor =
-    this->bulletHinge2->getRotationalLimitMotor(_index);
+#ifndef LIBBULLET_VERSION_GT_282
+  btRotationalLimitMotor
+#else
+  btRotationalLimitMotor2
+#endif
+  *motor = this->bulletHinge2->getRotationalLimitMotor(_index);
   if (motor)
     return motor->m_loLimit;
 
